apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -227,38 +233,51 @@ external_binary_to_mess_id(BExtMessID) when is_binary(BExtMessID) ->
 %% -----------------------------------------------------------------------
 %% XML
 
+%% @doc Adds all arcid elements (for backward compatibility)
```

Comment:
We may consider as well to deprecate `archived` element together with v0.2 support.